htsget-http-actix/Cargo.toml

```diff
@@ -9,13 +9,13 @@ default = ["async", "htsget-search/default", "htsget-http-core/default"]
 async = ["tokio", "futures", "htsget-search/async", "htsget-http-core/async"]
 
 [dependencies]
-actix-web = { version = "4.0.0" }
-envy = "0.4"
-serde = { version = "1", features = ["derive"] }
-serde_json = "1"
-futures-util = { version = "0.3.5", default-features = false }
+actix-web = "4.0.1"
+envy = "0.4.2"
+serde = { version = "1.0.136", features = ["derive"] }
```

Comment:
I think we should start leaving this simply as `1.0` so that we don't have to bump minor versions continuously. IIRC cargo will download the latest minor within that major?

Reply:
I changed this one so that it would match the other dependencies, which also specify patch versions (rightmost number). I believe [cargo](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio) would still download updates until 2.0.0?

I'm not really sure what the best style is here, whether we should specify only major/minor, or all version information. With some dependencies we use newer features, so it makes sense to specify more clearly which version it is. However, we could just do this on a case-by-case basis.

Reply:
Ah, sorry, I meant patch versions! Since 2.x would most probably be an "API breaking change", you don't want cargo to upgrade from 1.x to 2.x automatically. But back to my point, we should stop at minor numbers (as a general rule) and let the patches be, since they are pulled automatically, i.e:

```
(base) rvalls@m1 cargo_version_check % cargo build
    Updating crates.io index
   Compiling serde v1.0.136
   Compiling cargo_version_check v0.1.0 (/Users/rvalls/tmp/cargo_version_check)
    Finished dev [unoptimized + debuginfo] target(s) in 6.57s
(base) rvalls@m1 cargo_version_check % cat Cargo.toml
[package]
name = "cargo_version_check"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
serde = "1.0"
```

Reply:
Sure, that makes sense. What about the other crates that use patch versions? For example we have:
```Rust
tokio = { version = "1.17.0", features = ["full"], optional = true }
```
or:
```Rust
futures = { version = "0.3.21", optional = true }
```

Reply:
Same story: cut until minor version sans patch and cargo will keep collecting the patches without us changing it over time. As per `semver`, those unattended updates should not break functionality/APIs:

```
(base) rvalls@m1 cargo_version_check % cargo build
(...)
   Compiling tokio v0.3.7
   Compiling cargo_version_check v0.1.0 (/Users/rvalls/tmp/cargo_version_check)
    Finished dev [unoptimized + debuginfo] target(s) in 4.42s

(base) rvalls@m1 cargo_version_check % grep tokio Cargo.toml
tokio = "0.3"
```

Less babysitting work, leveraging [semver.org](https://semver.org/), essentially ;)